client/app/(protected)/(draft)/(teamBuilder)/players.tsx

```diff
@@ -20,33 +21,63 @@ import Table from "@/components/Table/Table";
 import TeamActionButtons from "@/components/TeamActionButtons";
 import TeamBudget from "@/components/TeamBudget";
 import { NbaPlayersController } from "@/controllers/nbaPlayersController";
+import { NbaTeamsController } from "@/controllers/nbaTeamsController";
 import { useAppDispatch, useAppSelector } from "@/state/hooks";
 import {
   addPlayerToLineup,
   removePlayerFromLineup,
   resetToSavedTeam,
 } from "@/state/slices/teamSlice";
+import { NbaTeam } from "@/types/nbaTeams";
 import { Player } from "@/types/players";
-import { FlexPosition, UTIL_POSITIONS } from "@/types/team";
+import { FlexPosition, PositionOption, UTIL_POSITIONS } from "@/types/team";
 import { isPlayerInLineup, resetTeamLineup } from "@/utils/teamUtils";
 
 type FetchPlayersParams = {
   pageParam?: FirebaseFirestoreTypes.DocumentSnapshot;
 };
 
+export type Filters = {
+  selectedTeams: NbaTeam[];
+  selectedPositions: PositionOption[];
+  salaryRange: { min: number; max: number };
+};
+
 const PAGE_SIZE = 25;
 
 const Players = () => {
   const [query, setQuery] = useState("");
+  const [teams, setTeams] = useState<NbaTeam[]>([]);
+  const [filters, setFilters] = useState<Filters>({
+    selectedTeams: [],
+    selectedPositions: [],
+    salaryRange: { min: 1000000, max: 150000000 },
+  });
   const team = useAppSelector((state) => state.team);
   const userId = useAppSelector((state) => state.user.id);
   const dispatch = useAppDispatch();
   const router = useRouter();
 
+  const [showFiltersDrawer, setShowFiltersDrawer] = useState(false);
+
+  const activeFilters =
```

Comment:
This isn't a blocker, but do you think the filter count should be on individual filters selected or categories selected? I have seen apps only do categories selected

Reply:
I guess it is more intuitive to be thinking "I have 2 filters applied" or "I am filtering by team and players, rather than saying I am filtering by these two teams and these two positions".

I will make the adjustment to count for filters being applied, similar to what I used with the salary range.


```suggestion
    const activeFilters =
        (filters.selectedTeams.length > 0 ? 1 : 0) +
        (filters.selectedPositions.length > 0 ? 1 : 0) +
        (filters.salaryRange.min !== 1000000 ||
        filters.salaryRange.max !== 150000000
            ? 1
            : 0);
```

Reply:
Yeah, I like it!